### PR TITLE
Add Product Admin for Store owners.

### PR DESCRIPTION
**Pull request by kjcioffi**
* Implement Login / Logout functionality in the base template navigation bar.
* Add Product Admin to the base template navigation bar.
* Implement Product Admin main base.
* Add update and delete page for managing existing products.
* Implemented a page for adding new products.

### CODE REVIEW

File: store/views.py [hunk 1]
```diff
@@ -86,21 +111,78 @@ def add_to_bag(request) -> JsonResponse:
...
+    def get_context_data(self, **kwargs: Any) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        context["store"] = self.request.user.store_set.get()
+        return context
+
+
+@login_required
+def product_admin_modify(request, pk):
+    product = get_object_or_404(Product, pk=pk)
```
Comment (ryaustin) [inline store/views.py:153]: Hmm, here I would also check that `store__owner=request.user` or include a conditional such as: 
```
if product.store.owner != request.user:
        return HttpResponseForbidden("You do not have permission to modify this product.")
```
- But I believe the first way is more ideal and reduces additional db queries.

File: store/views.py [hunk 2]
```diff
@@ -86,21 +111,78 @@ def add_to_bag(request) -> JsonResponse:
...
+
+
+class ProductAdminAdd(LoginRequiredMixin, CreateView):
+    form_class = ProductAdminForm
+    template_name = "store/product_admin_add.html"
+    success_url = reverse_lazy("store:product_admin")
+
+    def form_valid(self, form):
+        product = form.save(commit=False)
+        product.store = Store.objects.get(owner=self.request.user)
```
Comment (ryaustin) [inline store/views.py:186]: again here we see where a potential albeit rare issue could occur if a user creates multiple stores.
Seeing this twice now causes me to believe that creating a queryset manager could be ideal in this case such as `Store.objects.for_user()` and deal with the logic that returns the first store there. A queryset manager would be even more ideal for products.

**Review by ryaustin — CHANGES_REQUESTED** [review]
Just a couple of requested changes to address potential issues off the happy path and one security bug where a user could potentially delete a product from a store that is not theirs.

File: store/views.py (hunk 2 above)
Comment (kjcioffi) [reply store/views.py:190]: I love it, thanks for suggestion. I'll make use of custom managers with this logic.

File: store/views.py [hunk 3]
```diff
@@ -86,21 +111,78 @@ def add_to_bag(request) -> JsonResponse:
...
+    login_url = "/accounts/login"
+
+    def get_queryset(self) -> QuerySet[Product]:
+        # return products only from the store they own
+        return Product.objects.filter(store__owner=self.request.user)
+
+    def get_context_data(self, **kwargs: Any) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        context["store"] = self.request.user.store_set.get()
```
[the remark this answers is not in the document]
Comment (kjcioffi) [reply store/views.py]: A custom manager now handles this logic.

**Review by ryaustin — APPROVED** [review]
Wow, a lot of good work went into this PR
- queryset managers are just awesome, we see them used commonly in SaaS products.
- Good to include those tests with the updated functionality.
- Everything is working as intended with multiple stores for the same user.

**Comment by kjcioffi** [comment]
> Wow, a lot of good work went into this PR
> 
> * queryset managers are just awesome, we see them used commonly in SaaS products.
> * Good to include those tests with the updated functionality.
> * Everything is working as intended with multiple stores for the same user.

Thanks for the positive feedback! I hadn’t had an opportunity to use queryset managers in a practical sense yet. I wasn’t aware they were commonly used in SaaS products! I’m glad to hear that the tests for the updated functionality are effective and that everything is working smoothly with multiple stores for the same user.

**Pull request merged**